### ISSUE TITLE
Disconnecting and not reconnecting

### ISSUE THREAD

**Issue by AykutSarac** [timotejroiko/discord.js-light#33]
I'm getting that error in all of sudden while it's working fine.
```
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR Caught exception: TypeError: Cannot read property 'guild' of undefined
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR TypeError: Cannot read property 'guild' of undefined
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at ChannelManager.remove (/home/vcap/deps/0/node_modules/discord.js/src/managers/ChannelManager.js:44:17)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at GuildDeleteAction.client.actions.GuildDelete.handle (/home/vcap/deps/0/node_modules/discord.js-light/actions.js:126:15)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at Object.PacketHandlers.GUILD_DELETE (/home/vcap/deps/0/node_modules/discord.js-light/handlers.js:106:47)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at WebSocketManager.client.ws.handlePacket (/home/vcap/deps/0/node_modules/discord.js-light/actions.js:38:28)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at WebSocketShard.onPacket (/home/vcap/deps/0/node_modules/discord.js/src/client/websocket/WebSocketShard.js:444:22)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at WebSocketShard.onMessage (/home/vcap/deps/0/node_modules/discord.js/src/client/websocket/WebSocketShard.js:301:10)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at WebSocket.onMessage (/home/vcap/deps/0/node_modules/ws/lib/event-target.js:132:16)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at WebSocket.emit (node:events:378:20)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at Receiver.receiverOnMessage (/home/vcap/deps/0/node_modules/ws/lib/websocket.js:825:20)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at Receiver.emit (node:events:378:20)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at Receiver.dataMessage (/home/vcap/deps/0/node_modules/ws/lib/receiver.js:428:14)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at Receiver.getData (/home/vcap/deps/0/node_modules/ws/lib/receiver.js:367:17)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at Receiver.startLoop (/home/vcap/deps/0/node_modules/ws/lib/receiver.js:143:22)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at Receiver._write (/home/vcap/deps/0/node_modules/ws/lib/receiver.js:78:10)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at writeOrBuffer (node:internal/streams/writable:400:12)
   2021-02-09T16:45:54.69+0300 [APP/PROC/WEB/0] ERR     at _write (node:internal/streams/writable:341:10)
   2021-02-09T16:46:53.80+0300 [APP/PROC/WEB/0] OUT Debug log: [WS => Shard 0] [HeartbeatTimer] Didn't receive a heartbeat ack last time, assuming
zombie connection. Destroying and reconnecting.
   2021-02-09T16:46:53.80+0300 [APP/PROC/WEB/0] OUT     Status          : READY
   2021-02-09T16:46:53.80+0300 [APP/PROC/WEB/0] OUT     Sequence        : 66582
   2021-02-09T16:46:53.80+0300 [APP/PROC/WEB/0] OUT     Connection State: OPEN
   2021-02-09T16:46:53.80+0300 [APP/PROC/WEB/0] OUT Debug log: [WS => Shard 0] [DESTROY]
   2021-02-09T16:46:53.80+0300 [APP/PROC/WEB/0] OUT     Close Code    : 4009
   2021-02-09T16:46:53.80+0300 [APP/PROC/WEB/0] OUT     Reset         : true
   2021-02-09T16:46:53.80+0300 [APP/PROC/WEB/0] OUT     Emit DESTROYED: true
   2021-02-09T16:46:53.80+0300 [APP/PROC/WEB/0] OUT Debug log: [WS => Shard 0] Clearing the heartbeat interval.
```

**Comment by timotejroiko** [comment]
This happens if `client.channels` and `guild.channels` go out of sync, which should never happen by itself, but can happen if the user manually removes a channel from one but forgets to remove it from the other.

discord.js master has a fix for this, but they did not backport it to v12, i guess i can backport it myself later today

**Comment by AykutSarac** [comment]
Can you please elaborate _``if the user manually removes a channel from one but forgets to remove it from the other``_ part? I'm fetching channels that way ``message.guild.channels.fetch()`` and once I'm done, doing ``message.client.sweepChannels(0)`` to remove channels from cache if that's the problem. I don't know what else could be causing it to happen. Thanks for the response!

**Comment by timotejroiko** [comment]
like if one were to do guild.channels.cache.delete(id) instead of using client.sweepChannels(). But it can happen in other cases that i have not been able to figure out why yet. regardless, adding the check to ChannelManager.remove will fix it.

by the way, if you're sweeping the channel right after using it, you might as well use .fetch(id, false) to bypass the cache altogether, that way you dont need to sweep it.

I'll add the fix to .remove() soon

**Comment by timotejroiko** [comment]
released the fix in v3.5.3, can you test?

**Comment by AykutSarac** [comment]
> released the fix in v3.5.3, can you test?

Thanks, I've updated now. I will edit the post tomorrow with the result.
